File: JavaScript/encryption.ts
const encryption = (s: string) => {
  const s1 = s.replace(/\s+/g, '');
  const colLength = Math.ceil(Math.sqrt(s1.length));
  const columns = new Array(colLength).fill('');

  for (let i = 0; i < s1.length; i += 1) {
    columns[i % colLength] += s1[i];
  }

  return columns.join(' ');
};

[
  'have a nice day',
  'feed the dog      ',
  'chill out',
  'if man was meant to stay on the ground god would have given us roots',
].forEach((s) => {
  console.log(`encryption('${s}') = '${encryption(s)}'`);
});

File: Python/encryption.py
from math import ceil, sqrt
import timeit


def encryption(s: str) -> str:
    s = "".join(s.split(" "))
    col_len = ceil(sqrt(len(s)))
    cols = [[] for _ in range(col_len)]
    for i, ch in enumerate(s):
        cols[i % col_len].append(ch)

    cols = ["".join(li) for li in cols]
    return " ".join(cols)


def main():
    print(f"{encryption('have a nice day') = }")
    print(f"{encryption('feed the dog      ') = }")
    print(f"{encryption('chill out') = }")
    print(
        f"{encryption('if man was meant to stay on the ground god would have given us roots') = }"
    )
    print(
        "encryption('haveaniceday') =",
        max(
            timeit.repeat(
                stmt="encryption('haveaniceday')",
                setup="from __main__ import encryption",
                number=100_000,
            )
        ),
    )


if __name__ == "__main__":
    main()
